beacon-chain/rpc/validator_server.go

```diff
@@ -88,3 +88,30 @@ func (vs *ValidatorServer) ValidatorEpochAssignments(
 		},
 	}, nil
 }
+
+// ValidatorCommitteeAtSlot gets the committee at a certain slot where a validator's index is contained.
+func (vs *ValidatorServer) ValidatorCommitteeAtSlot(ctx context.Context, req *pb.CommitteeRequest) (*pb.CommitteeResponse, error) {
+	beaconState, err := vs.beaconDB.State()
+	if err != nil {
+		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
+	}
+	crossLinkCommittees, err := helpers.CrosslinkCommitteesAtSlot(beaconState, req.Slot, false /* registry change */)
+	if err != nil {
+		return nil, fmt.Errorf("could not get crosslink committees at slot %d: %v", req.Slot, err)
+	}
+	var committee []uint64
+	var shard uint64
+	for _, com := range crossLinkCommittees {
+		for _, i := range com.Committee {
+			if i == req.ValidatorIndex {
+				committee = com.Committee
+				shard = com.Shard
+				break
```

Comment:
Wouldn't this only break out of the inner loop ? it would still continue iterating over the committees in the outerloop